misc/spreading_model.py

```diff
@@ -0,0 +1,73 @@
+import network_diffusion as nd
+
+
+def _get_seeding_budget_for_network(
+        net: nd.MultilayerNetwork, actorwise: bool = False
+) -> dict[str, dict[str, int]]:
+    """
+    Return initial number of actors in compartmental graph proper to ActorTieredMICModel.
+     
+    This method patches nd.models.CompartmentalGraph.get_seeding_budget_for_network. It is fixed for
+    the case that a single actor is a source of the diffusion. Hence, the seeding budget (according
+    to the way it is used in network_diffusion) always equals to 1 / number of actors.
+
+    :param net: network to compute initial compartment shares.
+    :param actorwise: a field required by the original get_seeding_budget_for_network
+
+    :return: a dictionary with the initial number of actors in each compartment 
+    """
+    return {
+        FixedBudgetMICModel.PROCESS_NAME: {
+            FixedBudgetMICModel.INACTIVE_NODE: net.get_actors_num() - 1,
+            FixedBudgetMICModel.ACTIVE_NODE: 1,
+            FixedBudgetMICModel.ACTIVATED_NODE: 0,
+        }
+    }
+
+
+class FixedBudgetMICModel(nd.models.MICModel):
```

Comment:
the intuition for creating a custom model is a fact the in `network_diffusion` budget is being provided in percentage of agents, so that the model is independent on the network. Since we're using super small budget in this project (a single actor) for networks containing at least 500 nodes that fraction is very small.

Hence, I've created a custom class, whose method to convert percentages into numbers of agents for given network is patched. As a result, it always returns the desired division of states at the initial state of the diffusion.